reflection/reflection.fbs

```diff
@@ -20,6 +20,7 @@ enum BaseType : byte {
     ULong,
     Float,
     Double,
+    Array,
```

Comment:
Adding it in the middle rather than at the end will break all reflection files out there.